modules/smtp/functions.php

```diff
@@ -61,3 +61,92 @@ function get_reply_type($request) {
         return false;
     }
 }
+
+/**
+ * @subpackage smtp/functions
+ */
+if (!hm_exists('send_scheduled_message')) {
+function send_scheduled_message($handler, $mailbox, $folder, $msg_id, $send_now = false) {    
+    $msg_headers = $mailbox->get_message_headers($folder, $msg_id);    
+    $mailbox_details = $mailbox->get_config();       
+
+    try {
+        if (empty($msg_headers['X-Schedule'])) {
+            return false;
+        }
+
+        if (new DateTime($msg_headers['X-Schedule']) <= new DateTime() || $send_now) {
+            $profile = Hm_Profiles::get($msg_headers['X-Profile-ID']);
+            if (!$profile) {
+                $profiles = Hm_Profiles::search('server', $mailbox_details['server']);
+
+                if (!$profiles) {
+                    Hm_Debug::add(sprintf('ERRCannot find profiles corresponding with MAILBOX server: %s', $mailbox_details['server']));
+                    return false;
+                }
+                $profile = $profiles[0];
+            }
+
+            $smtp = Hm_SMTP_List::connect($profile['smtp_id'], false);
```

Comment:
Now that EWS has been merged, we no longer use direct smtp connections. Please go through Mailbox class like other smtp modules do it.